src/olympia/core/utils.py

```diff
@@ -0,0 +1,18 @@
+import json
+import os
+
+
+def get_version_json():
+    root = os.path.join(os.path.dirname(os.path.dirname(__file__)), '..', '..')
+    version_json = os.path.join(root, 'version.json')
+    version = None
+
+    if os.path.exists(version_json):
+        try:
+            with open(version_json) as fobj:
+                contents = fobj.read()
+                version = json.loads(contents)
+        except Exception:
```

Comment:
we were catching particular exceptions before, but it's just `Exception` - are there more possible (but acceptable) exceptions?

Reply:
There are a lot of exceptions that `could` be raised when reading a file and loading json.. It didn't make sense to me to try and collect all of them so I think there are 2 options:

1. don't catch any errors here, just let callers to the get_version_json handle them
2. catch all errors and return empty when something goes wrong.
3. Technically could revert to just the 2 specific previous exceptions, but that leads to inconsistent behavior.

Maybe we could log them at least though, to prevent silent errors from occurring.

Reply:
I'm struggling to think of any legitimate errors (i.e. that we'd expect in normal operation) that aren't either the file isn't accessible or isn't valid json.  But if we want this to _always_ succeed as to not bork the wider checking function then I guess logging the exact error instead.

Reply:
Actually looked at it again, and probably you are right to keep specific errors and just add the json decoding error as that is the most likely to occur if invalid json is recieved.

I realized we no longer need the KeyError as we aren't reading keys from the returned dictionary. I also realised we can catch the json errors and print those and os errors while returning a none object to prevent manage.py from totally failing because the version.json is missing or invalid.

https://github.com/mozilla/addons-server/pull/22210/files#diff-a99bef1b3c2386995edaab55833c3ac9a0f1a97e9e2679bb9555d99c332082d5R15